Lib/json/tool.py

```diff
@@ -22,7 +22,7 @@ def main():
     parser = argparse.ArgumentParser(prog=prog, description=description)
     parser.add_argument('infile', nargs='?', type=argparse.FileType(),
                         help='a JSON file to be validated or pretty-printed')
-    parser.add_argument('outfile', nargs='?', type=argparse.FileType('w'),
+    parser.add_argument('outfile', nargs='?', default='-',
```

Comment:
I guess it make sense to initialize `default=sys.stdout` like I suggest in #11992 and you can define `type=str`.

For some obscure reasons I imagine that someone would one day use `-` as the output file name...

So I guess for these reasons that using `sys.stdout` is a more proper way to formalize what is an output file and what is an output on stdout asked by users.

Reply:
`-` is common to denote `stdin` or `stdout` in command line utilities.

Here is how `grep` handles it: https://git.savannah.gnu.org/cgit/grep.git/tree/src/grep.c#n1886

You can always use `./-` to give a file with an obscure filename:
```
➜  ~ grep foo -
foo
foo
bar
➜  ~ grep foo ./-
grep: ./-: No such file or directory
```